File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/ObjC/CRUD/CRUD/ViewControllers/CrudOperationsViewController.m

#import "CrudOperationsViewController.h"
#import "CodeSampleViewController.h"
#import "RetrieveViewController.h"

#define CREATE @"Create"
#define RETRIEVE @"Retrieve"
#define UPDATE @"Update"
#define DELETE @"Delete"

#define SHOW_SAMPLE @"ShowCodeSample"
#define SHOW_RETRIEVE @"ShowRetrieve"

@interface CrudOperationsViewController() {
    NSArray<NSString *> *operations;
}
@end

@implementation CrudOperationsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    operations = @[CREATE, RETRIEVE, UPDATE, DELETE];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return operations.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"CrudOperationCell" forIndexPath:indexPath];
    cell.textLabel.text = [operations objectAtIndex:indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NSString *operation = [operations objectAtIndex:indexPath.row];
    if ([operation isEqualToString:RETRIEVE]) {
        [self performSegueWithIdentifier:SHOW_RETRIEVE sender:nil];
    }
    else {
        [self performSegueWithIdentifier:SHOW_SAMPLE sender:indexPath];
    }
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:SHOW_RETRIEVE]) {
        RetrieveViewController *retrieveVC = [segue destinationViewController];
        retrieveVC.tableName = self.tableName;
    }
    else {
        CodeSampleViewController *codeSampleVC = [segue destinationViewController];
        codeSampleVC.tableName = self.tableName;
        codeSampleVC.operation = [operations objectAtIndex:((NSIndexPath *)sender).row];
    }
}

@end

File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/Swift/CRUD/CRUD/ViewControllers/CrudOperationsViewController.swift

import UIKit

class CrudOperationsViewController: UITableViewController {
    
    var tableName: String!
    
    private let CREATE = "Create"
    private let RETRIEVE = "Retrieve"
    private let UPDATE = "Update"
    private let DELETE = "Delete"

    private let SHOW_SAMPLE = "ShowCodeSample"
    private let SHOW_RETRIEVE = "ShowRetrieve"

    private var operations = [String]()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        operations = [CREATE, RETRIEVE, UPDATE, DELETE]
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return operations.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "CrudOperationCell", for: indexPath)
        cell.textLabel?.text = operations[indexPath.row]
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let operation = operations[indexPath.row]
        if operation == RETRIEVE {
            performSegue(withIdentifier: SHOW_RETRIEVE, sender: nil)
        }
        else {
            performSegue(withIdentifier: SHOW_SAMPLE, sender: indexPath)
        }
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == SHOW_RETRIEVE {
            if let retrieveVC = segue.destination as? RetrieveViewController {
                retrieveVC.tableName = tableName
            }
        }
        else {
            if let codeSampleVC = segue.destination as? CodeSampleViewController,
                let indexPath = sender as? IndexPath {
                codeSampleVC.tableName = tableName
                codeSampleVC.operation = operations[indexPath.row]
            }
        }
    }
}
